src/app/views/infoblock/infoblock.js

```diff
@@ -0,0 +1,48 @@
+class InfoblockController {
+
+  /** @ngInject */
+  constructor($log) {
+    this.$log = $log;
+  }
+
+  $onChanges() {
+    if (!this.variable) {
+      return;
+    }
+
+    this.value = this.variable.value;
+    this.title = this.variable.title;
+
+    this.colorClass = '';
```

Comment:
Why not just eliminate all this logic, and instead have the component take a `colorClass` one-way bound attribute and let the parent controller pass in the name of the class to use?

I suppose you still have to provide isPercent and isChange so that the values can display properly.